src/renderer/entities/governance/model/referendum.ts

```diff
@@ -1,74 +1,77 @@
 import { type ApiPromise } from '@polkadot/api';
 import { createEffect, createEvent, createStore, sample } from 'effector';
-import { interval, readonly } from 'patronum';
+import { readonly } from 'patronum';
 
 import { type Chain, type ChainId, type Referendum, type ReferendumId } from '@/shared/core';
-import { addUnique } from '@shared/lib/utils';
+import { merge, nonNullable } from '@shared/lib/utils';
 import { governanceService } from '../lib/governanceService';
-
-type RequestListParams = {
-  chain: Chain;
-  api: ApiPromise;
+import { governanceSubscribeService } from '../lib/governanceSubscribeService';
+import { createSubscriber } from '../utils/createSubscriber';
+
+const mergeReferendums = (a: Referendum[], b: Referendum[]) => {
+  return merge(
+    a,
+    b,
+    (x) => x.referendumId,
+    (a, b) => parseInt(a.referendumId) - parseInt(b.referendumId),
+  );
 };
```

Comment:
Are you sure we need to merge them? We receive a full new updated list from the network anyway. In the case of merging, it can be a situation when the `referendum.id` is the same but `referendum.status` for example is different. In that case referendum list won't be updated when the `ongoing` referendum becomes `completed`

Reply:
Merging lists is safer option regardless of current implementation of subscribe method. Merge method overrides record by key (`referendumId` in this case), so it shouldn't be a problem.